include/circt/Dialect/RTL/RTLStatements.td

```diff
@@ -19,9 +19,9 @@ def ConnectOp : RTLOp<"connect", [InOutTypeConstraint<"src", "dest">]> {
     ```
     }];
 
-  let arguments = (ins InOutType:$dest, RTLValueType:$src);
+  let arguments = (ins Arg<InOutType, "", [MemWrite]>:$dest, RTLValueType:$src);
```

Comment:
I would recommend dropping the MemWrite/Read attributes.  They aren't used by your patch, so they are untested code.  Also, while I get what you're going for here, it isn't clear to me that this is the right way to model this.  It might be, it just hurts my brain ;-)

Reply:
They are being used to get CSE to delete read-only wires.  Whether that should just be folded into the explicit code is a different question.